CSharp/DevVmPowershell/Helpers/Implementations/InstanceSettingsHelper.cs

```diff
@@ -48,33 +74,72 @@ public bool UpdateInstanceSettingValue(string name, string section, string newVa
 					{
-						Relativity.Services.Result<InstanceSetting> result = instanceSettingQueryResultSet.Results.First();
-						if (result.Artifact != null)
+						objectType = new Helpers.RequestModels.ObjectType
+						{
+							Name = Constants.InstanceSetting.INSTANCE_SETTING_OBJECT_TYPE
+						},
+						fields = new object[]
 						{
-							Console.WriteLine("Successfully found the existing Instance Setting");
-							InstanceSetting instanceSetting = result.Artifact;
-							if (instanceSetting.Name == "ESIndexCreationSettings")
+							new
 							{
-								string previousValue = instanceSetting.Value;
-								newValue = previousValue.Replace("\"number_of_shards\": 12", "\"number_of_shards\": 2");
-								newValue = newValue.Replace("\"number_of_replicas\": 2", "\"number_of_replicas\": 0");
+								Name = "Name"
+							},
+							new
+							{
+								Name = "Value"
 							}
-							instanceSetting.Value = newValue;
-							instanceSettingManager.UpdateSingleAsync(instanceSetting).Wait();
-							Console.WriteLine("Successfully updated the Instance Setting");
-							return true;
+						},
+						condition = $"'Section' == '{section}' AND 'Name' == '{name}'"
+					},
+					start = 1,
+					length = 10
+				};
+
+				string queryRequest = JsonConvert.SerializeObject(objectManagerQueryRequestModel);
+				HttpResponseMessage queryResponse = RestHelper.MakePost(httpClient, queryUrl, queryRequest);
+				if (!queryResponse.IsSuccessStatusCode)
+				{
+					throw new Exception("Failed to Query for Agent Artifact Ids");
+				}
+				string result = queryResponse.Content.ReadAsStringAsync().Result;
+				JObject jObject = JObject.Parse(result);
+				int totalCount = jObject["TotalCount"].Value<int>();
+				if (totalCount > 0)
+				{
+					int instanceSettingArtifactId = jObject["Objects"][0]["ArtifactID"].Value<int>();
+					string instanceSettingName = jObject["Objects"][0]["FieldValues"][0]["Value"].Value<string>();
+					if (instanceSettingName == "ESIndexCreationSettings")
+					{
+						string instanceSettingValue = jObject["Objects"][0]["FieldValues"][1]["Value"].Value<string>();
+						newValue = instanceSettingValue.Replace("\"number_of_shards\": 12", "\"number_of_shards\": 2");
+						newValue = newValue.Replace("\"number_of_replicas\": 2", "\"number_of_replicas\": 0");
+					}
+					string updateUrl = $"Relativity.REST/api/Relativity.InstanceSettings/workspace/{Constants.EDDS_WORKSPACE_ARTIFACT_ID}/instancesettings/";
+					InstanceSettingManagerUpdateRequest instanceSettingManagerUpdateRequest = new InstanceSettingManagerUpdateRequest
+					{
+						instanceSetting = new RequestModels.InstanceSetting
+						{
+							ArtifactId = instanceSettingArtifactId,
+							Value = newValue
 						}
+					};
+					string updateRequest = JsonConvert.SerializeObject(instanceSettingManagerUpdateRequest);
+					HttpResponseMessage updateResponse = RestHelper.MakePut(httpClient, updateUrl, updateRequest);
+					if (updateResponse.IsSuccessStatusCode)
```

Comment:
we will need an else block here which throws an exception when status code is not success

- [ ] Add else block